kernprof.py

```diff
@@ -35,6 +35,100 @@ def execfile(filename, globals=None, locals=None):
             exec_(compile(f.read(), filename, 'exec'), globals, locals)
 # =====================================
 
+""" TODO: support python 2? altough it is depreciated """
+def execfile_inject(filename, globals=None, locals=None, include=None, exclude=None, 
+                        find="if __name__ == '__main__':", before=False, module=None):
+    """
+    Args:
+        include (str):   comma separated list of function/class names to include in profiling
+        exclude (str):   comma separated list of function/class names to exclude from profiling
+        find (str):      text to look for in code to add profiling code before or after
+                         default: "if __name__ == '__main__':"
+        before (bool):   if true, inser profiling code before find string, else after
+                         default: False
+        module (string): module/class/function name in script file to profile
+    """
+    import builtins
+    import re
+    exec_ = getattr(builtins, "exec")
+
+    include_ = ',include="{}"'.format(include) if include else ''
+    exclude_ = ',exclude="{}"'.format(exclude) if exclude else ''
+    module_ = '{}'.format(module) if module else ''
+    
+    with open(filename, 'r') as f:
+        code = f.read()
+        """ skip adding auto profiling code if already present and profiling all, or has include/exclude """
+        if code.find('decorate_with(profile)()') == -1 and code.find('decorate_with(profile,') == -1:
+            """ profiling code must be added after all declarations but before script is run
+            defaults to adding code after "if __name__ == '__main__':" as that is usually where
+            script execution is. 
+            """
+            needle = str(find) if find is not None else "if __name__ == '__main__':"
+            loc_1 = code.rfind(needle)
+            if loc_1 != -1:
+                loc_0 = code.rfind('\n',0,loc_1)+1
+                loc_2 = code.find('\n',loc_0)-1
+                if before:
+                    loc = loc_0
+                else:
+                    loc = loc_2+1
+                line = code[loc_0:loc_2+1]
+                """ match indentation of script when adding before or after """
+                indentation = re.match(r"\s*", line).group()
+                if not before and code[loc_2] == ':':
+                    indentation += '    '
+                """ profiling code that wraps modules/functions/classes to profile """
+                decorator_code = """
+{0}def decorate_with(decorator,include=None,exclude=None):
```

Comment:
please use `textwrap.dedent(text).strip('\n')` to maintain consistent indentation within the parent function.